System/Console/ANSI/Codes.hs

```diff
@@ -92,6 +94,17 @@ colorToCode color = case color of
   Cyan    -> 6
   White   -> 7
 
+
+-- | 'color8ToCode' @color@ returns the index of the color
+color8ToCode :: Color8 -> Color8Code
+color8ToCode (Gray8Color g)       = Color8Code $ 232 + clamp g 0 23
+color8ToCode (RGB8Color r' g' b') = Color8Code $ 16 + 36 * r + 6 * g + b
```

Comment:
Please add an explanation or reference for how/why this works

Reply:
I'll add some comments, it works because `0x10-0xE7:  6 × 6 × 6 cube (216 colors): 16 + 36 × r + 6 × g + b (0 ≤ r, g, b ≤ 5)` (from https://en.wikipedia.org/wiki/ANSI_escape_code)